File: code-examples/java-rs/usage-examples/changeStream.java
// Required MongoDB Java Reactive Streams Driver dependencies
import com.mongodb.ConnectionString;
import com.mongodb.reactivestreams.client.MongoClients;
import com.mongodb.reactivestreams.client.MongoClient;
import com.mongodb.reactivestreams.client.MongoCollection;
import com.mongodb.reactivestreams.client.MongoDatabase;
import com.mongodb.client.model.changestream.ChangeStreamDocument;
import org.bson.Document;
import reactor.core.publisher.Flux;

public class WatchCollectionChanges {

    // Replace the uri string with your MongoDB deployment's connection string
    private static final String CONNECTION_STRING = "<connection string uri>";

    public static void main(final String[] args) {
        MongoClient mongoClient = MongoClients.create(new ConnectionString(CONNECTION_STRING));

        try {
            MongoDatabase database = mongoClient.getDatabase("insertDB");
            MongoCollection<Document> haikus = database.getCollection("haikus");

            // Create a flux i.e watch change stream on the "haikus" collection
            Flux<ChangeStreamDocument<Document>> changeStreamFlux = Flux.from(haikus.watch());

            // Print change events as they occur
            changeStreamFlux.doOnNext(change -> System.out.println("Received change: " + change)).blockLast();

        } finally {
            // Close the MongoDB client connection
            mongoClient.close();
        }
    }
}

File: code-examples/scala/usage-examples/changeStream.scala
import org.mongodb.scala._

import scala.concurrent.Await
import scala.concurrent.duration._

// Replace the uri string with your MongoDB deployment's connection string.
val uri = "<connection string uri>"

val mongoClient: MongoClient = MongoClient(uri)

// Open a database and get the collection 'haikus'
val haikus = mongoClient.getDatabase("insertDB").getCollection("haikus")

// Create the change stream observable
val changeStreamObservable = haikus.watch()

// Define the observer
val observer = new Observer[ChangeStreamDocument[Document]] {
  override def onNext(result: ChangeStreamDocument[Document]): Unit = 
    println(s"Received change: $result")
    
  override def onError(e: Throwable): Unit = 
    System.err.println(s"Error: $e")
    
  override def onComplete(): Unit = 
    println("Change Stream completed")
}

// Subscribe the observer to the change stream observable
changeStreamObservable.subscribe(observer)

// Keep alive for a while to receive changes. Adjust according to use case.
Await.result(changeStreamObservable.toFuture(), 30.seconds)

// Close the MongoDB client connection
mongoClient.close()
